### PR TITLE
112 handle dynamic parameters

### LINKED ISSUE #112: Handle dynamic parameters
**Short description:**
Add the possibilty to have dynamic parameters.

**Describe the solution you'd like**
Example : An user send a message in a channel. We want to send by email the content, the channel, the guild and the author of the message.

### PR DESCRIPTION
**Pull request by ZQUEMA**
# Purpose

The purpose of this pull request is to handle the dynamic parameters as variable in the reaction.

Fixes #112 

## Added features

* New Model to store dynamic parameters
* Reaction can now use the dynamic parameters of a linked action.

## Updated features

* Action model has now dynamic parameters

### CODE REVIEW

File: server/prisma/schema.prisma [hunk 1]
```diff
@@ -84,22 +85,30 @@ model Parameter {
   ReactionParameters ReactionParameter[]
 }
 
+model DynamicParameter {
+  id          String  @id @unique @default(cuid())
+  name        String
+  description String?
+  Action      Action  @relation(fields: [actionId], references: [id])
```
Comment (LucasTesnier) [inline server/prisma/schema.prisma]: Missing onDelete and onUpdate field

File: server/services/discord/actions/on_join_voice_channel.js [hunk 2]
```diff
@@ -21,7 +21,10 @@ client.on('voiceStateUpdate', (oldChannel, newChannel) => {
       }
     ]
 
-    AreaGlue('DSC-02', parametersList)
+    const dynamicParameters = [
```
Comment (LucasTesnier) [inline server/services/discord/actions/on_join_voice_channel.js]: Format the code

**Review by LucasTesnier — CHANGES_REQUESTED** [review]
Some minor things to fix.

Is the Delete area work with the new Database scheme ?

File: server/prisma/schema.prisma (hunk 1 above)
Comment (ZQUEMA) [reply server/prisma/schema.prisma]: Yes, it does

File: server/services/discord/actions/on_join_voice_channel.js (hunk 2 above)
Comment (ZQUEMA) [reply server/services/discord/actions/on_join_voice_channel.js]: Done

**Review by LucasTesnier — DISMISSED** [review]

**Review by tbellicha — APPROVED** [review]

**Review by LucasTesnier — APPROVED** [review]

**Pull request merged**